blockchain/blockchain.go

```diff
@@ -854,4 +865,36 @@ func (bc *blockchain) replaceHeightAndHash(blk *Block) (uint64, hash.Hash32B, er
 	return tipHeight, tipHash, nil
 }
 
+func (bc *blockchain) SubscribeBlockCreation(ch chan *Block) error {
+	if bc.blocklistener == nil {
+		bc.blocklistener = []chan *Block{ch}
+	} else {
+		bc.blocklistener = append(bc.blocklistener, ch)
+	}
+	return nil
+}
+
+func (bc *blockchain) UnSubscribeBlockCreation(ch chan *Block) error {
+	for i, handler := range bc.blocklistener {
+		if ch == handler {
```

Comment:
Have you tested chan comparison will work?

BTW, it's better to have a test case to unsubscribe